### ISSUE TITLE
[choose keyboard]: when entering letters, keep adding them one-after-another to jump to an entry

### ISSUE THREAD

**Issue by star-buck** [calamares/calamares-todos#29]
Example:
Entering the letters G-E-R right after each other should jump to the entry Germany, but currently jumps to Georgia as first "G" entry, then E and then to first R letter entry.

**Comment by teo** [comment]
Looking into it.

**Comment by teo** [comment]
I've figured out the cause, and it appears to be a bug (or missing functionality) in QListWidget. QListWidget is supposed to behave the same way as a QListView plus a QAbstractListModel or similar, but this doesn't seem to be the case as far as `QAbstractItemView::keyboardSearch` is concerned. This function is what implements the type-to-search functionality we need here.
I don't have a fix yet, but I've been working on it. I've ran a quick and dirty test, and using QListView instead of QListWidget fixes this. Going from this test to the whole fix is not trivial, but I don't expect it to be hard either, it's just a matter of putting in the time to rewrite that page with a proper model+view instead of QListWidget.
